docs/snippets/angular/button-story-component-decorator.ts.mdx

```diff
@@ -0,0 +1,31 @@
+```ts
+// button.stories.ts
+
+import { Meta, Story } from '@storybook/angular';
+import { ButtonComponent } from './button.component';
+
+export default {
+  title: 'Example/Button',
+  component: ButtonComponent,
+  decorators: [
+    (storyFunc) => {
+      const story = storyFunc();
+
+      return {
+        ...story,
+        template: `<div style="margin: 3em">${story.template}</div>`,
+      };
+    },
+  ],
+} as Meta;
+
+// Note: A template with the component's selector is required in the component story for the decorators to take place.
+
+export const Default: Story<ButtonComponent> = () => ({
+  component: ButtonComponent,
+  moduleMetadata: {
+    declarations: [ButtonComponent],
+  },
+  template: '<app-button label="Submit"></app-button>',
+});
```

Comment:
@profanis we probably could use the `Template` itself instead of the actual story for the component, turning this snippet into:
```ts
// button.stories.ts

// same code as before

const Template: Story< ButtonComponent > = (args: ButtonComponent) => ({
  component: ButtonComponent,
  moduleMetadata: {
    declarations: [ButtonComponent],
  },
  props: args,
  template: '<app-button label="Submit"></storybook-button>',
});
```

Reasoning behind this is due to the fact that this snippet will be applied to the component's story so probably not a bad idea to change it into a familiar code base, as when you or any other Storybook user installs Storybook he'll see a familiar code base.

Reply:
I couldn't agree more with this.

I did the change

Reply:
@profanis i saw that you applied the changes and we'll need to take care of one item. The template string that i mentioned above is actually incorrect, it should be:
```js
template: '<app-button label="Submit"></app-button>',
````

Instead of 
```js
 template: '<app-button label="Submit"></storybook-button>',
```

Sorry for the misleading. Let me know when you've made the adjustments and this is good to go.

Stay safe

Reply:
my bad that I didn't notice this :/ 

Fixed